### PR TITLE
[12.0][FIX] hr_timesheet_sheet: unlink only generated AAL

### PR DESCRIPTION
**Pull request by alexey-pelykh**
Unlinking non-generated ALLs are very bad

### CODE REVIEW

File: hr_timesheet_sheet/models/hr_timesheet_sheet.py [hunk 1]
```diff
@@ -361,7 +361,7 @@ def unlink(self):
         analytic_timesheet_toremove = self.env['account.analytic.line']
         for sheet in self:
             analytic_timesheet_toremove += \
-                sheet.timesheet_ids.filtered(lambda t: not t.task_id)
+                sheet.timesheet_ids.filtered(lambda t: t.name == '/')
```
Comment (pedrobaeza) [inline hr_timesheet_sheet/models/hr_timesheet_sheet.py:364]: Isn't a more determinant condition?

Comment (alexey-pelykh) [reply hr_timesheet_sheet/models/hr_timesheet_sheet.py:364]: Unfortunately no, it's used all across the file

Comment (pedrobaeza) [reply hr_timesheet_sheet/models/hr_timesheet_sheet.py:364]: I think we need to set a specific sheet_id or something for avoiding ghost removals. Although of course your code is better than now.

Comment (alexey-pelykh) [reply hr_timesheet_sheet/models/hr_timesheet_sheet.py:364]: `sheet.timesheet_ids` is already linked by `sheet_id`

Comment (pedrobaeza) [reply hr_timesheet_sheet/models/hr_timesheet_sheet.py:364]: Then I don't get it... is this for some kind of cleanup?

Comment (alexey-pelykh) [reply hr_timesheet_sheet/models/hr_timesheet_sheet.py:364]: Yes, when AALs are created from matrix, they make no sense w/o sheet, yet manually created ones should remain

Comment (pedrobaeza) [reply hr_timesheet_sheet/models/hr_timesheet_sheet.py:364]: OK, then if this condition is harder enough, good to go.

**Review by pedrobaeza — APPROVED** [review]

File: hr_timesheet_sheet/models/hr_timesheet_sheet.py (hunk 1 above)
Comment (alexey-pelykh) [reply hr_timesheet_sheet/models/hr_timesheet_sheet.py:364]: Yeah, since I've just erased couple of manual entries because of this :)

**Comment by alexey-pelykh** [comment]
@aheficent I guess this one is for you as well

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: agree

**Pull request merged**